downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/GenerateGOAnnotationFile/MolecularFunctionAnnotationBuilder.java

```diff
@@ -0,0 +1,176 @@
+                    // For MF annotations, gets all proteins from the ActiveUnit if it exists or from the PhysicalEntity if not.
+                    GKInstance activeUnitInst = (GKInstance) catalystInst.getAttributeValue(ReactomeJavaConstants.activeUnit);
+                    GKInstance entityInst = activeUnitInst == null ? (GKInstance) catalystInst.getAttributeValue(ReactomeJavaConstants.physicalEntity) : activeUnitInst;
+                    Set<GKInstance> proteinInstances = getMolecularFunctionProteins(entityInst);
+                    processProteins(proteinInstances, reactionInst, catalystInst);
+                } else {
+                    logger.info("Invalid catalyst, skipping GO annotation");
+                }
+        }
+    }
+
+    /**
+     * This checks validity of a catalyst in a few ways. It first checks that the Catalyst PhysicalEntity is valid.
+     * Then it will check if the ActiveUnit is valid if it exists. If there is no ActiveUnit, the PhysicalEntity
+     * is checked to see if it is 'multi-instance', which will invalidate it.
+     * Catalysts with more than 1 ActiveUnit are also considered invalid.
+     * @param catalystInst -- GKInstance, catalyst instance taken from the original reaction instance.
+     * @return -- True if the catalyst instance has been deemed valid, false if not.
+     * @throws Exception -- MySQLAdaptor exception.
+     */
+    private static boolean validateMolecularFunctionCatalyst(GKInstance catalystInst) throws Exception {
+        GKInstance catalystPEInst = (GKInstance) catalystInst.getAttributeValue(ReactomeJavaConstants.physicalEntity);
+        boolean validCatalystPE = GOAGeneratorUtilities.validateCatalystPE(catalystPEInst);
+        boolean catalystValidity = false;
+        if (validCatalystPE) {
+            List<GKInstance> activeUnitInstances = catalystInst.getAttributeValuesList(ReactomeJavaConstants.activeUnit);
+            if (activeUnitInstances.size() == 0) {
+                catalystValidity = GOAGeneratorUtilities.multiInstancePhysicalEntity(catalystPEInst.getSchemClass()) ? false : true;
+            } else if (activeUnitInstances.size() == 1) {
+                catalystValidity = validCatalystAU(activeUnitInstances.get(0));
+            }
+        }
+        if (!catalystValidity) {
+            logger.info("Invalid catalyst, skipping GO annotation");
+        }
+        return catalystValidity;
+    }
+
+    /**
+     *  Validity of ActiveUnit is dependant on it not being an EntitySet with non-EWAS members.
+     *  This rule used to also exclude Complex and Polymers, but it seems ActiveUnits can't be either of those.
+     * @param activeUnitInst -- GKInstance, active unit instance taken from the catalyst instance.
+     * @return -- True if the activeUnit instance has been deemed valid, false if not.
+     * @throws Exception -- MySQLAdaptor exception.
+     */
+    private static boolean validCatalystAU(GKInstance activeUnitInst) throws Exception {
```

Comment:
I was thinking about this and we might want to invert the check here and look for the valid options of an EWAS or an Entity Set with only EWAS members instead of looking for invalid options.  I just worry about something being marked as valid unintentionally if we don't explicitly exclude it

Reply:
So instead of innocent until proven guilty, you are suggesting that we are only looking to check if the instance is an EWAS or an EntitySet comprised only of EWAS'? Otherwise return false?

Reply:
This makes sense, I've made the change (in an upcoming commit)

Reply:
> So instead of innocent until proven guilty, you are suggesting that we are only looking to check if the instance is an EWAS or an EntitySet comprised only of EWAS'? Otherwise return false?

Yeah, exactly.  I like the innocent until proven guilty analogy, btw :).